digiwf-email-integration/src/main/java/io/muenchendigital/digiwf/email/integration/infrastructure/streaming/MessageProcessor.java

```diff
@@ -0,0 +1,72 @@
+package io.muenchendigital.digiwf.email.integration.infrastructure.streaming;
+
+import io.muenchendigital.digiwf.email.integration.domain.exception.MissingInformationMailException;
+import io.muenchendigital.digiwf.email.integration.domain.model.Mail;
+import io.muenchendigital.digiwf.email.integration.domain.service.MailingService;
+import io.muenchendigital.digiwf.spring.cloudstream.utils.api.streaming.infrastructure.RoutingCallback;
+import io.muenchendigital.digiwf.spring.cloudstream.utils.api.streaming.service.CorrelateMessageService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cloud.function.context.MessageRoutingCallback;
+import org.springframework.context.annotation.Bean;
+import org.springframework.mail.MailException;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class MessageProcessor {
+
+    public static final String TYPE_HEADER_SEND_MAIL_FROM_EVENT_BUS = "sendMailFromEventBus";
+    private final MailingService mailingService;
+    private final CorrelateMessageService correlateMessageService;
+    private static final String MAIL_SENT_STATUS = "mailSentStatus";
+
+    /**
+     * Override the custom router of the digiwf-spring-cloudstream-utils. We only have one type we need to map.
+     * @return the custom router
+     */
+    @Bean
+    public MessageRoutingCallback mailRouter() {
```

Comment:
would configure this in the starter configuration maybe someone wants to add own api endpoints with custom message handlers.